File: scripts/generate-download-qr.py
#!/usr/bin/env python3
"""Generate QR PNG pointing to the Amenallah Android download page (not the raw APK)."""

from __future__ import annotations

import argparse
import os
import sys

try:
    import qrcode
    from qrcode.constants import ERROR_CORRECT_M
except ImportError:
    print("Install: pip install 'qrcode[pil]'", file=sys.stderr)
    sys.exit(1)


def main() -> None:
    root = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
    default_url = os.environ.get("NEXT_PUBLIC_APP_URL", "http://localhost:3000").rstrip("/") + "/download"
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument(
        "--url",
        default=default_url,
        help="Full URL of the download page (default: NEXT_PUBLIC_APP_URL/download)",
    )
    parser.add_argument(
        "--out",
        default=os.path.join(root, "public", "images", "android-download-qr.png"),
        help="Output PNG path",
    )
    args = parser.parse_args()

    qr = qrcode.QRCode(version=None, error_correction=ERROR_CORRECT_M, box_size=10, border=2)
    qr.add_data(args.url)
    qr.make(fit=True)
    img = qr.make_image(fill_color="#0f172a", back_color="white")
    os.makedirs(os.path.dirname(args.out), exist_ok=True)
    img.save(args.out)
    print(f"Wrote {args.out}")
    print(f"Encodes: {args.url}")


if __name__ == "__main__":
    main()

File: scripts/generate-download-qr.mjs
#!/usr/bin/env node
/**
 * Generate QR PNG pointing to the Amenallah Android download page (not the raw APK).
 * Usage:
 *   NEXT_PUBLIC_APP_URL=https://example.tn node scripts/generate-download-qr.mjs
 *   node scripts/generate-download-qr.mjs --url https://example.tn/download
 */
import fs from "fs"
import path from "path"
import { fileURLToPath } from "url"
import QRCode from "qrcode"

const __dirname = path.dirname(fileURLToPath(import.meta.url))

const args = process.argv.slice(2)
const urlIdx = args.indexOf("--url")
const outIdx = args.indexOf("--out")
const base = (process.env.NEXT_PUBLIC_APP_URL || "http://localhost:3000").replace(/\/$/, "")
const url = urlIdx >= 0 && args[urlIdx + 1] ? args[urlIdx + 1] : `${base}/download`
const out =
  outIdx >= 0 && args[outIdx + 1]
    ? args[outIdx + 1]
    : path.join(__dirname, "..", "public", "images", "android-download-qr.png")

fs.mkdirSync(path.dirname(out), { recursive: true })
await QRCode.toFile(out, url, {
  margin: 2,
  width: 360,
  color: { dark: "#0f172a", light: "#ffffff" },
})
console.log(`Wrote ${out}`)
console.log(`Encodes: ${url}`)
